src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+    where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    where TOperationContext : JsonOperationContext
+{
+    public IndexHandlerProcessorForConvertAutoIndex([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    private string GetName() => RequestHandler.GetStringQueryString("name", required: true);
+
+    private ConvertType GetConvertType()
+    {
+        var typeAsString = RequestHandler.GetStringQueryString("type", required: true);
+
+        if (Enum.TryParse(typeAsString, ignoreCase: true, out ConvertType convertType) == false)
+            throw new InvalidOperationException($"Could not parse '{typeAsString}' to any known conversion type.");
+
+        return convertType;
+    }
+
+    public override async ValueTask ExecuteAsync()
+    {
+        var name = GetName();
+        var type = GetConvertType();
+
+        using (RequestHandler.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            var record = RequestHandler.ServerStore.Cluster.ReadRawDatabaseRecord(context, RequestHandler.DatabaseName);
+            if (record.AutoIndexes.TryGetValue(name, out var autoIndex) == false)
+                throw IndexDoesNotExistException.ThrowForAuto(name);
+
+            switch (type)
+            {
+                case ConvertType.Csharp:
+                    var result = AutoToStaticIndexConverter.Instance.ConvertToAbstractIndexCreationTask(autoIndex, out var csharpClassName);
+
+                    csharpClassName = $"{csharpClassName}.cs";
+
+                    HttpContext.Response.Headers[Constants.Headers.ContentDisposition] = $"attachment; filename=\"{csharpClassName}\"; filename*=UTF-8''{csharpClassName}";
+
+                    await using (var writer = new StreamWriter(RequestHandler.ResponseBodyStream()))
+                    {
+                        await writer.WriteLineAsync(result);
+                    }
+                    break;
+                case ConvertType.Export:
+                    await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+                    {
+                        var definition = AutoToStaticIndexConverter.Instance.ConvertToIndexDefinition(autoIndex);
+
+                        writer.WriteStartObject();
+
+                        writer.WritePropertyName("Indexes");
+                        writer.WriteStartArray();
+
+                        writer.WriteIndexDefinition(context, definition);
+
+                        writer.WriteEndArray();
+
+                        writer.WriteEndObject();
+                    }
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+
+            }
+        }
+    }
+
+    private enum ConvertType
+    {
+        Csharp,
```

Comment:
`Csharp_class` maybe to avoid the confusion with `CSharp` as the language parameter?